File: webapi/Core/Models/Exam/FlashCard.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ThoughtzLand.Core.Models.Common;

namespace ThoughtzLand.Core.Models.Exam
{

	public class FlashCard
	{
		public int id { get; set; }
		public int nodeId { get; set; }
		public string? question { get; set; }
		public string? description { get; set; }
		public IEnumerable<FlashCardAnswer>? answers { get; set; }
		public Language? language { get; set; }
		public int hitsInRow { get; set; }
		public int requiredHits { get; set; }
		public int totalHits { get; set; }
		public int level { get; set; }
		public DateTime nextExamDate { get; set; }
		public bool isCompleted { get; set; }
		public int questPrice { get; set; }
	}
}

File: client/src/app/Core/Models/flash-card.ts
import { IFlashCardAnswer } from "./flash-card-answer";
import { ILanguage } from "./language";

export interface IFlashCard{
  id: number;
  nodeId: number;
  question?: string;
  description?: string;
  answers?: IFlashCardAnswer[] | null; // Assuming FlashCardAnswer is a TypeScript type/interface
  language?: ILanguage | null; // Assuming Language is a TypeScript type/interface
  hitsInRow: number;
  requiredHits: number;
  totalHits: number;
  level: number;
  nextExamDate: Date;
  isCompleted: boolean;
  questPrice: number;
}

export interface ICreateFlashCardDto{
  nodeId: number;
  question?: string;
  description?: string;
  languageId: number;
  nextExamDate: Date;
  requiredHits: number;
}

export interface IUpdateFlashCardDto{
  id: number;
  question?: string | null;
  description?: string | null;
  languageId: number;
  //hitsInRow: number;
  //requiredHits: number;
  //totalHits: number;
  //nextExamDate: Date;
}
